zyte_common_items/pipelines.py

```diff
@@ -53,3 +55,67 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropLowProbabilityItemPipeline:
+    """This pipeline drops an item if its probability, defined in the settings,
+    is less than the specified threshold.
+
+    There are two ways that thresholds can be defined per item type using
+    the `ITEM_PROBABILITY_THRESHOLDS` setting:
+
+    With paths to Item classes:
+        ITEM_PROBABILITY_THRESHOLDS = {
+            "zyte_common_items.items.Article": 0.1,
+            "zyte_common_items.items.Product": 0.2
+        }
+
+    With Item classes:
+        from zyte_common_items import Article, Product
+        ITEM_PROBABILITY_THRESHOLDS = {
+            Article: 0.1,
+            Product: 0.2
+        }
+    """
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def init_thresholds(self, spider):
+        from scrapy.utils.misc import load_object
+
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = load_object(item) if isinstance(item, str) else item
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
+        return self.thresholds.get(type(item), DEFAULT_ITEM_PROBABILITY_THRESHOLD)
+
+    def get_item_name(self, item):
+        item_name = item.__class__.__name__
+        return item_name[0].lower() + item_name[1:]
```

Comment:
Very minor: I think I prefer the current case since it accurately reflects the item class name.

Reply:
We would have an inconsistency with the current e-commerce stat, and with scrapy-zyte-api (`scrapy-zyte-api/request_args/product`). But we all seem to agree on not trying to keep the current e-commerce stat, and since this is about items and not Zyte API params, the item class may indeed be the way to go.

However, from that point of view, I would expect the import path, i.e. 2 classes named `Product` can be 2 different classes.